internal/services/issuer/verifiable_credential_test.go

```diff
@@ -67,22 +116,180 @@ func (s *CredentialTestSuite) TestVerifiableCredential() {
+		VehicleNFT        models.VehicleNFT
+		AftermarketDevice models.AftermarketDevice
+	}{
+		{
+			Name:             "No corresponding aftermarket device for address",
+			ReturnsError:     true,
+			ExpectedResponse: "sql: no rows in result set",
+		},
+		{
+			Name:              "active credential",
+			ReturnsError:      false,
+			UserDeviceTable:   userDevice,
+			MetaTxTable:       metaTx,
+			VCTable:           credential,
+			VehicleNFT:        nft,
+			AftermarketDevice: aftermarketDevice,
+		},
+		{
+			Name:             "vin associated with device not confirmed",
+			ReturnsError:     true,
+			ExpectedResponse: "invalid vin",
+			UserDeviceTable: models.UserDevice{
+				ID:                 deviceID,
+				UserID:             userDeviceID,
+				DeviceDefinitionID: deiceDefID,
+				VinConfirmed:       false,
+				VinIdentifier:      null.StringFrom(vin),
+			},
+			MetaTxTable:       metaTx,
+			VCTable:           credential,
+			VehicleNFT:        nft,
+			AftermarketDevice: aftermarketDevice,
+		},
+		{
+			Name:            "inactive credential",
+			ReturnsError:    false,
+			UserDeviceTable: userDevice,
+			MetaTxTable:     metaTx,
+			VCTable: models.VerifiableCredential{
+				ClaimID:        claimID,
+				Credential:     []byte{},
+				ExpirationDate: time.Now().AddDate(0, 0, -10),
+			},
+			VehicleNFT:        nft,
+			AftermarketDevice: aftermarketDevice,
+		},
+	}
+
+	for _, c := range cases {
+		s.T().Run(c.Name, func(t *testing.T) {
+
+			err := c.UserDeviceTable.Insert(ctx, s.pdb.DBS().Writer, boil.Infer())
+			require.NoError(s.T(), err)
+
+			err = c.MetaTxTable.Insert(ctx, s.pdb.DBS().Writer, boil.Infer())
+			require.NoError(s.T(), err)
+
+			err = c.VCTable.Insert(ctx, s.pdb.DBS().Reader, boil.Infer())
+			require.NoError(s.T(), err)
+
+			err = c.VehicleNFT.Insert(ctx, s.pdb.DBS().Writer, boil.Infer())
+			require.NoError(s.T(), err)
+
+			err = c.AftermarketDevice.Insert(ctx, s.pdb.DBS().Writer, boil.Infer())
+			require.NoError(s.T(), err)
+
+			err = s.iss.Handle(s.ctx, &ADVinCredentialEvent{
+				CloudEvent: shared.CloudEvent[json.RawMessage]{
+					Data:    rawMsg,
+					Time:    time.Now(),
+					ID:      deviceID,
+					Subject: common.Bytes2Hex(ownerAddress.Bytes),
+				},
+				Signature: signature,
+			})
+
+			if c.ReturnsError {
+				assert.NotNil(s.T(), c.ExpectedResponse, err.Error())
+			} else {
+				require.NoError(s.T(), err)
+			}
+
+			_, err = models.AftermarketDevices().DeleteAll(ctx, s.pdb.DBS().Writer)
+			require.NoError(s.T(), err)
+
+			_, err = models.VehicleNFTS().DeleteAll(ctx, s.pdb.DBS().Writer)
+			require.NoError(s.T(), err)
+
+			_, err = models.VerifiableCredentials().DeleteAll(ctx, s.pdb.DBS().Writer)
+			require.NoError(s.T(), err)
+
+			_, err = models.MetaTransactionRequests().DeleteAll(ctx, s.pdb.DBS().Writer)
+			require.NoError(s.T(), err)
+
+			_, err = models.UserDevices().DeleteAll(ctx, s.pdb.DBS().Writer)
+			require.NoError(s.T(), err)
```

Comment:
Can use this helper to wipe everything:

https://github.com/DIMO-Network/devices-api/blob/5e958d9ba5230fb4f577f81d6d94ec493ce3e505/internal/test/helpers.go#L184